### PR TITLE
chore: update dependencies with breaking changes

### PR DESCRIPTION
**Pull request by DaniPopes**

### CODE REVIEW

File: crates/net/dns/src/resolver.rs
Comment (DaniPopes) [inline crates/net/dns/src/resolver.rs:1]: Only code change

File: crates/primitives/Cargo.toml [hunk 1]
```diff
@@ -48,11 +48,11 @@ tokio-stream.workspace = true
 bytes.workspace = true
 serde.workspace = true
 serde_json.workspace = true
-serde_with = "2.1.0"
+serde_with = "3.3.0"
 thiserror.workspace = true
-sucds = "0.5.0"
+sucds = "~0.6"
```
Comment (DaniPopes) [inline crates/primitives/Cargo.toml:53]: 0.7 contains a lot of breaking changes to the API, I'd rather not update this

File: crates/primitives/Cargo.toml [hunk 2]
```diff
@@ -68,23 +68,23 @@ sha2 = "0.10.7"
 triehash = "0.8"
 # See to replace hashers to simplify libraries
 plain_hasher = "0.2"
-hash-db = "0.15"
+hash-db = "~0.15"
```
Comment (DaniPopes) [inline crates/primitives/Cargo.toml:71]: Same

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**